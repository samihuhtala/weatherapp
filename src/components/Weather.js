import React, { Component } from 'react';

class Weather extends Component {
    constructor(props) {
        super(props);
        this.state = {temp: 0, city: '', listitem: [], displayc: ''};
    }

    fetchWeather = (e) => {
        e.preventDefault();
        fetch ('http://api.openweathermap.org/data/2.5/weather?q='+this.state.city+'&units=Metric&APPID=f3139d9f8a64cc9cf04a4b6425aaa543')
        .then(response => response.json())
        .then(responseData => {
            this.setState ({
                temp: responseData.main.temp,
                displayc: this.state.city,
            })
        })
        .catch(err => console.error(err));
       
    }
    handleChange = (e) => {
        this.setState({
            city: e.target.value
        });
    }
    addListitem = (e) => {
        e.preventDefault();
        const object = {displayc: this.state.displayc, temp: this.state.temp}
        this.setState({
            listitem: [...this.state.listitem, object]
        });
    }

    render() {
        return (
            <div>
                <h2>Search for city</h2>
                <input type="text" onChange={this.handleChange} />
                <input type ="submit" onClick={this.fetchWeather} value="Search" />
                <div>Temperature of {this.state.displayc} {this.state.temp}&#8451;</div>
                <h2>Save searched cities</h2>
                <input type ="submit" onClick={this.addListitem} value="Save"/>

                <table className="table">
                <thead>
                    <tr>
                        <td>City</td>
                        <td>Temperature</td>
                    </tr>
                </thead>
                <tbody>
                    {this.state.listitem.map((item,index) =>
                        <tr key ={index}>
                        <td>{item.displayc}</td>
                        <td>{item.temp}&#8451;</td>
                        </tr>
                        )}
                </tbody>
                </table>

            </div>
        );
    }
}

export default Weather;