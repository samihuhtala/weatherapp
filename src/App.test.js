import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import TestRenderer from 'react-test-renderer';
import Weather from './components/Weather';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('renders without crashing 2', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Weather />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const testRenderer = TestRenderer.create(<App />).toJSON();
  expect(testRenderer).toMatchSnapshot();
});
it('renders correctly', () => {
  const testRenderer = TestRenderer.create(<App page="localhost:3000">localhost</App>).toJSON();
  expect(testRenderer).toMatchSnapshot();
});
it('renders correctly part 2 ', () => {
  const testRenderer = TestRenderer.create(<Weather />).toJSON();
  expect(testRenderer).toMatchSnapshot();
});
it('renders correctly part 2 ', () => {
  const testRenderer = TestRenderer.create(<Weather page="localhost:3000">localhost</Weather>).toJSON();
  expect(testRenderer).toMatchSnapshot();
});