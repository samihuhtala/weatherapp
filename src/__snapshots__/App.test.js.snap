// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <h2>
      Search for city
    </h2>
    <input
      onChange={[Function]}
      type="text"
    />
    <input
      onClick={[Function]}
      type="submit"
      value="Search"
    />
    <div>
      Temperature of 
      
       
      0
      ℃
    </div>
    <h2>
      Save searched cities
    </h2>
    <input
      onClick={[Function]}
      type="submit"
      value="Save"
    />
    <table
      className="table"
    >
      <thead>
        <tr>
          <td>
            City
          </td>
          <td>
            Temperature
          </td>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="App"
>
  <div>
    <h2>
      Search for city
    </h2>
    <input
      onChange={[Function]}
      type="text"
    />
    <input
      onClick={[Function]}
      type="submit"
      value="Search"
    />
    <div>
      Temperature of 
      
       
      0
      ℃
    </div>
    <h2>
      Save searched cities
    </h2>
    <input
      onClick={[Function]}
      type="submit"
      value="Save"
    />
    <table
      className="table"
    >
      <thead>
        <tr>
          <td>
            City
          </td>
          <td>
            Temperature
          </td>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`renders correctly part 2  1`] = `
<div>
  <h2>
    Search for city
  </h2>
  <input
    onChange={[Function]}
    type="text"
  />
  <input
    onClick={[Function]}
    type="submit"
    value="Search"
  />
  <div>
    Temperature of 
    
     
    0
    ℃
  </div>
  <h2>
    Save searched cities
  </h2>
  <input
    onClick={[Function]}
    type="submit"
    value="Save"
  />
  <table
    className="table"
  >
    <thead>
      <tr>
        <td>
          City
        </td>
        <td>
          Temperature
        </td>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`renders correctly part 2  2`] = `
<div>
  <h2>
    Search for city
  </h2>
  <input
    onChange={[Function]}
    type="text"
  />
  <input
    onClick={[Function]}
    type="submit"
    value="Search"
  />
  <div>
    Temperature of 
    
     
    0
    ℃
  </div>
  <h2>
    Save searched cities
  </h2>
  <input
    onClick={[Function]}
    type="submit"
    value="Save"
  />
  <table
    className="table"
  >
    <thead>
      <tr>
        <td>
          City
        </td>
        <td>
          Temperature
        </td>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
